tests/ssg_test_suite/test_env.py

```diff
@@ -216,15 +219,34 @@ def __init__(self, mode, hypervisor, domain_name):
         self.hypervisor = hypervisor
         self.domain_name = domain_name
         self.snapshot_stack = None
+        self.keep_snapshots = keep_snapshots
 
         self._origin = None
 
+    def has_ssgts_prefix(self, snapshot_name):
+        if str(snapshot_name).startswith(common.SNAPSHOT_PREFIX):
+            return True
+        return False
+
+    def snapshot_lookup(self, snapshot_name):
+        return self.domain.snapshotLookupByName(snapshot_name)
+
+    def snapshots_cleanup(self):
+        snapshot_list = self.domain.snapshotListNames()
+        for snapshot_name in snapshot_list:
+            if self.has_ssgts_prefix(snapshot_name):
+                snapshot = self.snapshot_lookup(snapshot_name)
+                snapshot.delete()
```

Comment:
is it always guaranteed `snapshot` will be different a valid snapshot. Does it make sense to check for None here?

Reply:
If the snapshot is not found an exception will be raised by the `snapshotLookupByName` function. So, at this point there will be always a valid snapshot.